chain/network/src/peer_manager/network_state/routing.rs

```diff
@@ -37,11 +37,24 @@ impl NetworkState {
 
     /// Constructs a partial edge to the given peer with the nonce specified.
     /// If nonce is None, nonce is selected automatically.
-    pub fn propose_edge(&self, peer1: &PeerId, with_nonce: Option<u64>) -> PartialEdgeInfo {
+    pub fn propose_edge(
+        &self,
+        clock: &time::Clock,
+        peer1: &PeerId,
+        with_nonce: Option<u64>,
```

Comment:
is this even used now once you merged with master?

Reply:
yes it is - in last_edge.